puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -73,6 +75,41 @@ def build_events_list
     end
   end
 
+  # @return Array[Hash]
+  # @api private
+  def build_logs_list
+    profile("Build logs list (count: #{logs.count})",
+            [:puppetdb, :logs_list, :build]) do
+      log_list = []
+      logs.each do |log|
```

Comment:
Can you clarify this? log_hash is being pushed onto log_list at the end of the block and log_hash is difference from log, so I don't see how this works without another helper.